// storybook/dist/core-events/index.js
// src/core-events/index.ts
var R = /* @__PURE__ */ ((E) => (E.CHANNEL_WS_DISCONNECT = "channelWSDisconnect", E.CHANNEL_CREATED = "channelCreated", E.CONFIG_ERROR = "co\
nfigError", E.STORY_INDEX_INVALIDATED = "storyIndexInvalidated", E.STORY_SPECIFIED = "storySpecified", E.SET_CONFIG = "setConfig", E.SET_STORIES =
"setStories", E.SET_INDEX = "setIndex", E.SET_CURRENT_STORY = "setCurrentStory", E.CURRENT_STORY_WAS_SET = "currentStoryWasSet", E.FORCE_RE_RENDER =
"forceReRender", E.FORCE_REMOUNT = "forceRemount", E.PRELOAD_ENTRIES = "preloadStories", E.STORY_PREPARED = "storyPrepared", E.DOCS_PREPARED =
"docsPrepared", E.STORY_CHANGED = "storyChanged", E.STORY_UNCHANGED = "storyUnchanged", E.STORY_RENDERED = "storyRendered", E.STORY_FINISHED =
"storyFinished", E.STORY_MISSING = "storyMissing", E.STORY_ERRORED = "storyErrored", E.STORY_THREW_EXCEPTION = "storyThrewException", E.STORY_RENDER_PHASE_CHANGED =
"storyRenderPhaseChanged", E.STORY_HOT_UPDATED = "storyHotUpdated", E.PLAY_FUNCTION_THREW_EXCEPTION = "playFunctionThrewException", E.UNHANDLED_ERRORS_WHILE_PLAYING =
"unhandledErrorsWhilePlaying", E.UPDATE_STORY_ARGS = "updateStoryArgs", E.STORY_ARGS_UPDATED = "storyArgsUpdated", E.RESET_STORY_ARGS = "res\
etStoryArgs", E.SET_FILTER = "setFilter", E.SET_GLOBALS = "setGlobals", E.UPDATE_GLOBALS = "updateGlobals", E.GLOBALS_UPDATED = "globalsUpda\
ted", E.REGISTER_SUBSCRIPTION = "registerSubscription", E.PREVIEW_INITIALIZED = "previewInitialized", E.PREVIEW_KEYDOWN = "previewKeydown", E.
PREVIEW_BUILDER_PROGRESS = "preview_builder_progress", E.SELECT_STORY = "selectStory", E.STORIES_COLLAPSE_ALL = "storiesCollapseAll", E.STORIES_EXPAND_ALL =
"storiesExpandAll", E.DOCS_RENDERED = "docsRendered", E.SHARED_STATE_CHANGED = "sharedStateChanged", E.SHARED_STATE_SET = "sharedStateSet", E.
NAVIGATE_URL = "navigateUrl", E.UPDATE_QUERY_PARAMS = "updateQueryParams", E.REQUEST_WHATS_NEW_DATA = "requestWhatsNewData", E.RESULT_WHATS_NEW_DATA =
"resultWhatsNewData", E.SET_WHATS_NEW_CACHE = "setWhatsNewCache", E.TOGGLE_WHATS_NEW_NOTIFICATIONS = "toggleWhatsNewNotifications", E.TELEMETRY_ERROR =
"telemetryError", E.FILE_COMPONENT_SEARCH_REQUEST = "fileComponentSearchRequest", E.FILE_COMPONENT_SEARCH_RESPONSE = "fileComponentSearchRes\
ponse", E.SAVE_STORY_REQUEST = "saveStoryRequest", E.SAVE_STORY_RESPONSE = "saveStoryResponse", E.ARGTYPES_INFO_REQUEST = "argtypesInfoReque\
st", E.ARGTYPES_INFO_RESPONSE = "argtypesInfoResponse", E.CREATE_NEW_STORYFILE_REQUEST = "createNewStoryfileRequest", E.CREATE_NEW_STORYFILE_RESPONSE =
"createNewStoryfileResponse", E))(R || {}), S = R, {
  CHANNEL_WS_DISCONNECT: _,
  CHANNEL_CREATED: T,
  CONFIG_ERROR: A,
  CREATE_NEW_STORYFILE_REQUEST: O,
  CREATE_NEW_STORYFILE_RESPONSE: N,
  CURRENT_STORY_WAS_SET: r,
  DOCS_PREPARED: I,
  DOCS_RENDERED: D,
  FILE_COMPONENT_SEARCH_REQUEST: o,
  FILE_COMPONENT_SEARCH_RESPONSE: C,
  FORCE_RE_RENDER: P,
  FORCE_REMOUNT: L,
  GLOBALS_UPDATED: Y,
  NAVIGATE_URL: a,
  PLAY_FUNCTION_THREW_EXCEPTION: U,
  UNHANDLED_ERRORS_WHILE_PLAYING: H,
  PRELOAD_ENTRIES: W,
  PREVIEW_INITIALIZED: G,
  PREVIEW_BUILDER_PROGRESS: d,
  PREVIEW_KEYDOWN: i,
  REGISTER_SUBSCRIPTION: t,
  RESET_STORY_ARGS: p,
  SELECT_STORY: l,
  SET_CONFIG: F,
  SET_CURRENT_STORY: y,
  SET_FILTER: e,
  SET_GLOBALS: c,
  SET_INDEX: h,
  SET_STORIES: f,
  SHARED_STATE_CHANGED: g,
  SHARED_STATE_SET: u,
  STORIES_COLLAPSE_ALL: s,
  STORIES_EXPAND_ALL: x,
  STORY_ARGS_UPDATED: V,
  STORY_CHANGED: m,
  STORY_ERRORED: M,
  STORY_INDEX_INVALIDATED: Q,
  STORY_MISSING: w,
  STORY_PREPARED: B,
  STORY_RENDER_PHASE_CHANGED: X,
  STORY_RENDERED: b,
  STORY_FINISHED: q,
  STORY_SPECIFIED: K,
  STORY_THREW_EXCEPTION: Z,
  STORY_UNCHANGED: z,
  STORY_HOT_UPDATED: j,
  UPDATE_GLOBALS: k,
  UPDATE_QUERY_PARAMS: J,
  UPDATE_STORY_ARGS: $,
  REQUEST_WHATS_NEW_DATA: n,
  RESULT_WHATS_NEW_DATA: v,
  SET_WHATS_NEW_CACHE: EE,
  TOGGLE_WHATS_NEW_NOTIFICATIONS: RE,
  TELEMETRY_ERROR: SE,
  SAVE_STORY_REQUEST: _E,
  SAVE_STORY_RESPONSE: TE,
  ARGTYPES_INFO_REQUEST: AE,
  ARGTYPES_INFO_RESPONSE: OE
} = R;
export {
  AE as ARGTYPES_INFO_REQUEST,
  OE as ARGTYPES_INFO_RESPONSE,
  T as CHANNEL_CREATED,
  _ as CHANNEL_WS_DISCONNECT,
  A as CONFIG_ERROR,
  O as CREATE_NEW_STORYFILE_REQUEST,
  N as CREATE_NEW_STORYFILE_RESPONSE,
  r as CURRENT_STORY_WAS_SET,
  I as DOCS_PREPARED,
  D as DOCS_RENDERED,
  o as FILE_COMPONENT_SEARCH_REQUEST,
  C as FILE_COMPONENT_SEARCH_RESPONSE,
  L as FORCE_REMOUNT,
  P as FORCE_RE_RENDER,
  Y as GLOBALS_UPDATED,
  a as NAVIGATE_URL,
  U as PLAY_FUNCTION_THREW_EXCEPTION,
  W as PRELOAD_ENTRIES,
  d as PREVIEW_BUILDER_PROGRESS,
  G as PREVIEW_INITIALIZED,
  i as PREVIEW_KEYDOWN,
  t as REGISTER_SUBSCRIPTION,
  n as REQUEST_WHATS_NEW_DATA,
  p as RESET_STORY_ARGS,
  v as RESULT_WHATS_NEW_DATA,
  _E as SAVE_STORY_REQUEST,
  TE as SAVE_STORY_RESPONSE,
  l as SELECT_STORY,
  F as SET_CONFIG,
  y as SET_CURRENT_STORY,
  e as SET_FILTER,
  c as SET_GLOBALS,
  h as SET_INDEX,
  f as SET_STORIES,
  EE as SET_WHATS_NEW_CACHE,
  g as SHARED_STATE_CHANGED,
  u as SHARED_STATE_SET,
  s as STORIES_COLLAPSE_ALL,
  x as STORIES_EXPAND_ALL,
  V as STORY_ARGS_UPDATED,
  m as STORY_CHANGED,
  M as STORY_ERRORED,
  q as STORY_FINISHED,
  j as STORY_HOT_UPDATED,
  Q as STORY_INDEX_INVALIDATED,
  w as STORY_MISSING,
  B as STORY_PREPARED,
  b as STORY_RENDERED,
  X as STORY_RENDER_PHASE_CHANGED,
  K as STORY_SPECIFIED,
  Z as STORY_THREW_EXCEPTION,
  z as STORY_UNCHANGED,
  SE as TELEMETRY_ERROR,
  RE as TOGGLE_WHATS_NEW_NOTIFICATIONS,
  H as UNHANDLED_ERRORS_WHILE_PLAYING,
  k as UPDATE_GLOBALS,
  J as UPDATE_QUERY_PARAMS,
  $ as UPDATE_STORY_ARGS,
  S as default
};
